### PR TITLE
Adding Compile time reducer support

### PR DESCRIPTION
**Pull request by ya1gaurav**
Include file cotire.cmake

### CODE REVIEW

**Comment by ya1gaurav** [comment]
Refer - http://unriskinsight.blogspot.in/2014/09/sharpen-your-tools.html
Cotire will reduce compile time for cmake project, many open source projects are adopting it.

**Comment by cdunn2001** [comment]
I'm not sure about this yet. It seems to generate a unity build. But if that has value, we could do that ourself, right?

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +66,7 @@ jsoncpp_parse_version( ${JSONCPP_VERSION} JSONCPP_VERSION )
 #ENDIF(NOT JSONCPP_VERSION_FOUND)
 SET( JSONCPP_SOVERSION 11 )
 SET( JSONCPP_USE_SECURE_MEMORY "0" CACHE STRING "-D...=1 to use memory-wiping allocator for STL" )
+SET (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/CMake")
```
Comment (okodron) [inline CMakeLists.txt]: I'm not sure it's a good idea to override the default ``CMAKE_MODULE_PATH`` value.
You can still include ``cotire.cmake`` relatively to the currently processed ``CmakeLists.txt``:
```
INCLUDE(${CMAKE_CURRENT_SOURCE_DIR}/cotire.cmake)
```

**Comment by hjmjohnson** [comment]
This issue has become stale due to inactivity.  If you want this issue to be reviewed again, please update the PR resolving the merge conflicts and re-open this issue.

Closing until updated information is provided.

**Pull request closed without merging**